src/lib/default.nix

```diff
@@ -104,7 +104,10 @@
   # }
   prepareSourceTreeInternal = sourceRoot: relPath: name: depth: let
     relPath' = relPath;
-    fullPath' = "${sourceRoot}/${relPath}";
+    fullPath' = l.path {
+      path = "${sourceRoot}/${relPath}";
+      inherit name;
+    };
```

Comment:
Do we really need `builtins.path` here? Is there any context inherited from here.
I think we should be careful using `builtins.path`, and make sure that we only use it in places were we really need it.
It is an expensive operation. AFAIU nix copied the whole path each time this is called.

If we really need this on the main source, then it should be done only once, not several times.

Reply:
Yeah, I was just looking at that. This and the one at sanitizePath doesn't seem to be needed, so I removed them.